### PR TITLE
docs: Basic cookbook template using quarto

### PR DESCRIPTION
**Pull request by rodriguesfred**
### Summary
Created a basic folder structure for a Quarto cookbook.
Created an action to render and deploy cookbook to gihub pages

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔧 Refactoring (no functional changes, no api changes)
- [ ] ⚡ Performance improvement
- [x] 🧪 Test addition or improvement
- [ ] 🔨 Build/CI changes
- [ ] 🎨 Style/formatting changes

## Changes Made
Created a basic folder structure for a Quarto cookbook.
Created an action to render and deploy cookbook to gihub pages

## Testing
Confirmed the cookbook deploys to github pages on [https://best-practice-and-impact.github.io/afcharts-py/]

## Documentation
- [ ] Documentation updated
- [x] README updated
- [ ] API documentation updated
- [ ] No documentation changes needed

**Documentation Changes:**
Added a section onto the readme on how to test the cookbook

## Dependencies
- [ ] No new dependencies
- [x] New dependencies added (list below)
- [ ] Dependencies updated (list below)
- [ ] Dependencies removed (list below)

**Dependency Changes:**
Requires Quarto installed locally to test the cookbook.

### Code Quality
- [ ] Code follows the project's style guidelines
- [ ] Self-review of code completed
- [ ] Code is properly commented
- [ ] No unnecessary console.log/print statements
- [ ] No commented-out code (unless intentionally documented)

### Testing & Quality Assurance
- [ ] Tests pass locally
- [ ] Tests added for new functionality
- [ ] Edge cases considered and tested
- [ ] Error handling implemented where appropriate

**Github issue link**:  #_\<Enter GitHub issue ref here\>_

## Describe your changes

_\<Enter a description for this PR here\>_

## Checklist before requesting a review

The PR raiser/assignee must ensure that for:
- All PRs:
    - [ ] The [**branching strategy**] (branching of the `main` branch) has been followed.
    - [ ] A **self-review** of the changes has been performed by the author.
    - [ ] Any **merge conflicts** have been resolved (see bottom of PR).
    -  All PR **workflow checks** have passed (see outcomes at bottom of PR):
        - [ ] The [**Conventional Commit**](https://www.conventionalcommits.org/en/v1.0.0) messages format has been followed for the PR title (see README.md).
        - [ ] All **linting issues** raised by MegaLinter have been resolved.
        - [ ] All **tests** have passed.
- New features or bugfixes (less relevant for `doc`,`style`, `chore` etc.):
    - [ ] Thorough **tests** have been added/updated for modified code.
    - [ ] Updates to **documentation** have been made as necessary.

### CODE REVIEW

**Comment by rodriguesfred** [comment]
Could someone review? So we can close this PR and #26

File: src/afcharts/cookbook/getting-started.qmd [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+To install the package:
+
+```bash
+pip install 'afcharts_py'
+```
+
+**use_afcharts** <br>
+The examples in this cookbook use the afcharts theme and colour functions explicitly, however it may be easier to make use of the `use_afcharts()` function if your charts all require a similar style. More information on `use_afcharts` can be found on the [homepage](https://best-practice-and-impact.github.io/afcharts/#use-afcharts-as-default).
```
Comment (nickhenden) [inline src/afcharts/cookbook/getting-started.qmd:20]: Remove for now. Not sure if we'll have a function like this or what the syntax will be

File: src/afcharts/cookbook/getting-started.qmd [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+**Note on use of titles, subtitles and captions** <br>
+Titles, subtitles and captions have been embedded in the charts in this cookbook for demonstration purposes. However, for accessibility reasons, it is usually preferable to provide titles in the body of the page rather than embedded within the image of the plot.
+
+The following packages are required to produce the example charts in this cookbook:
```
Comment (nickhenden) [inline src/afcharts/cookbook/getting-started.qmd:25]:
```suggestion
**Dependencies** <br>
The following packages are required to produce the example charts in this cookbook:
```

File: src/afcharts/cookbook/images/logo.svg
Comment (nickhenden) [inline src/afcharts/cookbook/images/logo.svg:1]: The logo is already in the root `assets/` folder, can we use that rather than duplicate it here?

File: src/afcharts/cookbook/04-colour-palettes.qmd [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+df = pd.DataFrame(list(test_palette.items()), columns=["Name", "Hex Code"])
+df
+
+df["Preview"] = df["Hex Code"].apply(
+    lambda x: f'<span style="background-color:{x};padding:5px 20px;display:inline-block;"></span>'
+)
+from IPython.display import display, HTML
+
+display(HTML(df.to_html(escape=False)))
+```
```
[the remark this answers is not in the document]
Comment (rodriguesfred) [reply src/afcharts/cookbook/04-colour-palettes.qmd:41]: its a good point. I think at the moment its written as raw code. Not as a script. One to figure out later but yes Quarto should be able to render.

File: .gitignore [hunk 4]
```diff
@@ -133,3 +133,8 @@ README_files/*
 # Cookbook temp docs
 .quarto/
 _book/
+*.html
+*.md
```
Comment (nickhenden) [inline .gitignore:137]: Remove `*.md`. Don't want to ignore the README etc.

Comment (nickhenden) [reply .gitignore:137]: You can do e.g. `src/afcharts/cookbook/*.md`

File: src/afcharts/cookbook/03-plotly-usage.qmd [hunk 5]
```diff
@@ -41,24 +41,22 @@ fig.update_layout(
...
-    meta=dict(
-        alt="Bar chart showing sales of Product A (23 units), Product B (45 units), and Product C (56 units)."
-    ),
+    meta=dict(alt="Bar chart showing sales of Product A (23 units), Product B (45 units), and Product C (56 units)."),
 )
 
 fig.show()
 ```
 
 ##/ Example of line chart with duo palette
```
Comment (nickhenden) [inline src/afcharts/cookbook/03-plotly-usage.qmd:50]: Why the forward slash? It doesn't seem to render as a heading. Should just be
```suggestion
## Example of a line chart with duo palette
```

File: src/afcharts/cookbook/03-plotly-usage.qmd [hunk 6]
```diff
@@ -101,12 +99,64 @@ fig.update_layout(
...
-    ),
+    meta=dict(alt="Bar chart showing sales of Product A (23 units), Product B (45 units), and Product C (56 units)."),
 )
 
 fig.update_yaxes(range=[0, 82])
 
 fig.show()
 ```
+
+##/ Example of scatter chart
```
Comment (nickhenden) [inline src/afcharts/cookbook/03-plotly-usage.qmd:110]:
```suggestion
## Example of a scatter chart
```

File: src/afcharts/cookbook/_quarto.yml [hunk 7]
```diff
@@ -5,7 +5,7 @@ project:
 book:
   title: "afcharts-py cookbook"
   repo-branch: main
-  favicon: images/logo.svg
+  favicon: ../assets/logo.svg
```
Comment (nickhenden) [inline src/afcharts/cookbook/_quarto.yml:8]: Logo doesn't seem to be rendering now sadly :/ At least, not on https://best-practice-and-impact.github.io/afcharts-py/04-colour-palettes.html

File: src/afcharts/cookbook/index.qmd [hunk 8]
```diff
@@ -5,14 +5,11 @@ description-meta: |
 
 
 ::: {.content-visible when-format="html"}
-# Welcome {.unnumbered}
-This cookbook provides examples of the code used to produce various chart types using `afcharts_py`. There are also examples to demonstrate how to apply further customisation to `afcharts_py` charts.
+# Welcome <img src="../assets/logo.svg" alt="afcharts logo" style="float: right; height: 150px;"> {.unnumbered}
```
Comment (nickhenden) [inline src/afcharts/cookbook/index.qmd:8]: Logo also doesn't render here

File: tests/plotly_scatter_test.py [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+import plotly.graph_objects as go
+import plotly.io as pio
+
+from afcharts.assets.af_colours import duo
+
+# AF Package
+from afcharts.theme_af import theme_af
+
+
+def test_plolty_line_chart():
```
Comment (nickhenden) [inline tests/plotly_scatter_test.py:13]:
```suggestion
def test_plotly_line_chart():
```

**Review by nickhenden — COMMENTED** [review]
Charts look great! Though sadly the logo doesn't render.

File: src/afcharts/cookbook/04-colour-palettes.qmd (hunk 3 above)
[the remark this answers is not in the document]
Comment (nickhenden) [reply src/afcharts/cookbook/04-colour-palettes.qmd:43]: The charts now render (yay) but this bit of code with the colours doesn't :/

**Comment by rodriguesfred** [comment]
@nickhenden after several issues deploying that were actually due to github systems were down I've addressed all the comments but struggling with the logo outside the quarto folder.

I got the colours qmd to render the colours
